src/Utilities/Workloads.js

```diff
@@ -144,3 +145,48 @@ export const pruneWorkloadsRulesFilters = (localFilters, filterCategories) => {
     return arr;
   }, []);
 };
+
+export const switchSort = (sortIndex, item) => {
+  const rule = item[0]?.rule; // Using optional chaining to check if 'rule' is defined
+  switch (sortIndex) {
+    case 1:
+      return rule?.details || '';
+    case 2:
+      return rule?.total_risk || [];
+    case 3:
+      return rule?.objects?.length || 0;
+    case 4:
+      return rule?.modified || '';
+    default:
+      return 0;
+  }
+};
+
+export const sortWithSwitch = (
+  sortIndex,
+  sortDirection,
+  filteredRows,
+  firstRule
+) => {
+  return sortIndex >= 0 && !firstRule
```

Comment:
Looking at switchSort: it looks like the sorting index starts from 1. I think we need to rewrite it to
```
  return sortIndex >= 1 && !firstRule
```
instead, WDYT?

Reply:
Updated it